tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
```

Comment:
from the [mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for)

I think it would make sense to refer to it as the `prohibit the use of let in for loop initialization`